src/mscorlib/src/System/Diagnostics/Stackframe.cs

```diff
@@ -12,244 +12,27 @@
-
-
-        internal virtual void SetMethodBase(MethodBase mb)
-        {
-            method = mb;
-        }
-
-        internal virtual void SetOffset(int iOffset)
-        {
-            offset = iOffset;
-        }
-
-        internal virtual void SetILOffset(int iOffset)
-        {
-            ILOffset = iOffset;
-        }
-
-        internal virtual void SetFileName(String strFName)
-        {
-            strFileName = strFName;
-        }
-
-        internal virtual void SetLineNumber(int iLine)
-        {
-            iLineNumber = iLine;
-        }
-
-        internal virtual void SetColumnNumber(int iCol)
-        {
-            iColumnNumber = iCol;
-        }
-
-        internal virtual void SetIsLastFrameFromForeignExceptionStackTrace(bool fIsLastFrame)
-        {
-            fIsLastFrameFromForeignExceptionStackTrace = fIsLastFrame;
-        }
-
-        internal virtual bool GetIsLastFrameFromForeignExceptionStackTrace()
-        {
-            return fIsLastFrameFromForeignExceptionStackTrace;
-        }
-
-        // Returns the method the frame is executing
-        // 
-        public virtual MethodBase GetMethod()
-        {
-            Contract.Ensures(Contract.Result<MethodBase>() != null);
-
-            return method;
-        }
-
-        // Returns the offset from the start of the native (jitted) code for the
-        // method being executed
-        // 
-        public virtual int GetNativeOffset()
-        {
-            return offset;
-        }
-
-
-        // Returns the offset from the start of the IL code for the
-        // method being executed.  This offset may be approximate depending
-        // on whether the jitter is generating debuggable code or not.
-        // 
-        public virtual int GetILOffset()
-        {
-            return ILOffset;
-        }
-
-        // Returns the file name containing the code being executed.  This 
-        // information is normally extracted from the debugging symbols
-        // for the executable.
-        //
-        public virtual String GetFileName()
-        {
-            return strFileName;
-        }
-
-        // Returns the line number in the file containing the code being executed.  
-        // This information is normally extracted from the debugging symbols
-        // for the executable.
-        //
-        public virtual int GetFileLineNumber()
-        {
-            return iLineNumber;
-        }
-
-        // Returns the column number in the line containing the code being executed.  
-        // This information is normally extracted from the debugging symbols
-        // for the executable.
-        //
-        public virtual int GetFileColumnNumber()
-        {
-            return iColumnNumber;
-        }
-
-
-        // Builds a readable representation of the stack frame
-        //
+        /// <summary>
+        /// Builds a readable representation of the stack frame
+        /// </summary>
         public override String ToString()
```

Comment:
Well, the problem is that in CoreCLR there's always a MethodBase available for each managed frame. That is not the case in .NET Native / CoreRT, there can be frames without reflection info available. At this point I made shared the two main components of ToString - FormatMethodNameUsingMethodInfo and FormatFileInfo - so that ToString became a few lines. Again, let's leave the final decision for the integration phase.